### ISSUE TITLE
Lag is a feature that should be removed.

### ISSUE THREAD

**Issue by Zardinio** [EarthMC/Issue-Tracker#1220]
The server is clearly struggling, for reasons that are outside my scope.
We have had consistent crashes on the server for seemingly no reason, despite the fact the server was doing quite fine not too long ago. What has changed since that time? Has adding 30 additional players pushed the server beyond what the server is capable of?

What are our options as players on this issue? Is there anything I can do to decrease the lag outside of not playing? Do we need limitations on hoppers per chunk? Should we limit item frames and in an indirect way:map art lag (without removing map art)
What are our options to handle the lag? What can we do?

I haven't heard anything from any of the staff. Sometimes they will be on when the server crashes and don't even acknowledge the crash, this has happened multiple times where vote-party gets reloaded like 3 hours back when it was 200 votes away.

This is such a great community and would hate to see people leave because of the lag.

Z

![image](https://user-images.githubusercontent.com/65791200/90449966-88ae1080-e0b6-11ea-9d38-924c742eb767.png)

**Comment by DeadGlamour** [comment]
10/10 for the title.
And ye it did become even worse recently, hopefuly the staff will at least say what causes this lag.

**Comment by Nikocholas** [comment]
I'm in the screenshot 😌

**Comment by JaVolimKatarinu** [comment]
The issue is the Towny plugin. Basically the way it is coded, plugin cant handle big number of players. I believe that the Towny team is working on improving performance of their plugin.
The second issue is Minecraft. 1.12 version was incredibly better than any of the newer versions released. [As you can see here](https://www.reddit.com/r/Minecraft/comments/bgs1fb/talking_performance_in_114/), Mojang brought new stuff into the game which has costed us a lot of ticks. I don't think there is anything that the Staff team can do, except removing even more "content" from the game, starting with Villagers. There is a huge issue with villagers since their new update, forcing the game to constantly check for their beds and workstations. That wouldn't be a problem if there weren't hundreds, maybe a thousand villagers present on the server.

**Comment by EmattpoYou** [comment]
The server has been somewhat stable on 1.15, but recently, it has been just so goddamn laggy for no reason. It's not the version, and the playercount has been continuously reduced, but lag hasnt stopped.

**Comment by suneettipirneni** [comment]
> The issue is the Towny plugin. Basically the way it is coded, plugin cant handle big number of players. I believe that the Towny team is working on improving the performance of their plugin.

This is absolutely incorrect, in fact, this cannot be further from the truth. In many of the timings reports given by towny server admins, Towny rarely even has an empirical presence. Towny even has been shown to have less of a footprint than other more widely used plugins such as Essentials (Even though essentials doesn't really cause that much lag as well). The main source of lag is from server software, and as stated many times before Minecraft servers are severely single-threaded, which means that they don't take full advantage of the processor cores.

> The second issue is Minecraft. 1.12 version was incredibly better than any of the newer versions released. As you can see here, Mojang brought new stuff into the game which has costed us a lot of ticks. I don't think there is anything that the Staff team can do, except removing even more "content" from the game, starting with Villagers. There is a huge issue with villagers since their new update, forcing the game to constantly check for their beds and workstations. That wouldn't be a problem if there weren't hundreds, maybe a thousand villagers present on the server.

While ticking the villagers may have contributed to the overall lag experienced in later versions of Minecraft, most developers can agree that the way chunks were loaded became significantly worse post-1.12. Chunk loading is one of the most expensive operations in Minecraft, and it is constantly done. So by making the chunk heavier, many slowdowns occur as a result. Adding on to my previous point, 1.16.2 added a lot of new content to the game, much more than 1.14 did, and yet it still didn't affect performance as much as 1.14 did. This is because new content is not *necessarily* bound to the speed of the server.

**Comment by JaVolimKatarinu** [comment]
> 
> This is absolutely incorrect, in fact, this cannot be further from the truth. In many of the timings reports given by towny server admins, Towny rarely even has an empirical presence. Towny even has been shown to have less of a footprint than other more widely used plugins such as Essentials (Even though essentials doesn't really cause that much lag as well). The main source of lag is from server software, and as stated many times before Minecraft servers are severely single-threaded, which means that they don't take full advantage of the processor cores.

Oh, I'm sorry but I was sure that I read something somewhere about Towny not behaving very well when it comes to the number of 100+ players on the server. I assumed that meant on lagg and possible crashes/safe modes. Anyway, my mistake. Please exclude my comment in further discussion.

**Comment by Nikocholas** [comment]
Do any of you know how does Hypixel make to handle 100.000 players in around 30 different games without lagging?

**Comment by suneettipirneni** [comment]
> Do any of you know how does Hypixel make to handle 100.000 players in around 30 different games without lagging?

Yes, it’s simple there are thousands of servers they host and the players are distributed throughout them.

 “Why can’t EMC do that?” 

Well for starters everyone would be divided up on separate “earths” defeating the geopolitical aspect. The edits made by players on one server not show on another. Also many plugins including towny do not support cross server communication as it retains a lot of information. This is a stark difference to Hypixel’s rather quick and trivial mini games. Hypixel can easily host a minigame on another server. You can’t just host the same towny world on a separate servers.

tldr; There are thousands of players on the hypixel  *network* not thousands on the *server*. On EMC everyone is playing in the same server so the situation/circumstances between EMC and Hypixel is completely different.

**Comment by KarlOfDuty** [comment]
As stated above the towny plugin has nothing to do with our performance issues. We are running a half finished server software and thus the performance gets pretty bad from time to time. We are looking to switch but there are currently some issues to overcome first.
